src/errors.rs

```diff
@@ -4,14 +4,11 @@
 // LICENSE file in the root directory of this source tree.
 
 //! A list of error types which are produced during an execution of the protocol
-use std::convert::Infallible;
-use std::error::Error;
-use std::fmt::Debug;
-
-use displaydoc::Display;
+use core::convert::Infallible;
+use core::fmt::{Debug, Display};
```

Comment:
We should leave the `Error` implementations, we but we should just hide it behind the `std` crate feature